contracts/Unibright.sol

```diff
@@ -149,22 +159,21 @@ contract UBTSplitter is Context, Ownable {
         );
 
         require(
-            whitelistedTokens[address(token)],
+            theWhitelistedToken == address(token),
             "UBTSplitter: not whitelisted"
         );
-
-        uint256 totalReceived = token.balanceOf(address(this)) +
-            erc20TotalReleased[token];
-        uint256 payment = _pendingPayment(
-            revenueAddress,
-            totalReceived,
-            erc20Released[token][revenueAddress]
-        );
+   
+        uint256 alreadyReceivedSinceLastPayeeUpdate = theErc20ReleasedPerRecipientInPeriods[theErc20CurrentPeriod][revenueAddress];
+        uint256 toBeReleased = theErc20ToBeReleasedInPeriod;        
```

Comment:
nitpick: 167 and 168 can be one line, it is more readable:
uint256 toBeReleased = theErc20ToBeReleasedInPeriod + theErc20NotReleasedInLastPeriod;